runtime/compilers/rillv1/parser.go

```diff
@@ -302,11 +300,22 @@ func (p *Parser) TrackedPathsInDir(dir string) []string {
 	return paths
 }
 
+// GetDotEnv returns all the project's environment variables
+func (p *Parser) GetDotEnv() map[string]string {
+	env := make(map[string]string)
+	for _, envMap := range p.DotEnv {
+		for k, v := range envMap {
+			env[k] = v
+		}
+	}
+	return env
+}
```

Comment:
Would be nice to make this deterministic in case the same env var is defined multiple places (bad thing to do, but non-determinism would make it trickier to debug) – e.g. sort the paths and add them in sorted order